Crash on startup, related to shortcut / update?

**Describe the crash**
The app crashes repeatedly on startup (it will not open anymore). I think it is related to the recent update.

**To Reproduce**
Unfortunately I am not sure what exactly caused it to happen.

**Stack trace**

```
12-04 08:14:23.887 17946 17946 E AndroidRuntime: java.lang.RuntimeException: Unable to create service fr.neamar.kiss.dataprovider.AppProvider: java.lang.NullPointerException: Attempt to read from field 'java.lang.String fr.neamar.kiss.db.ShortcutRecord.name' on a null object reference in method 'java.lang.String fr.neamar.kiss.utils.ShortcutUtil.generateShortcutId(fr.neamar.kiss.utils.UserHandle, fr.neamar.kiss.db.ShortcutRecord)'
12-04 08:14:23.887 17946 17946 E AndroidRuntime: Caused by: java.lang.NullPointerException: Attempt to read from field 'java.lang.String fr.neamar.kiss.db.ShortcutRecord.name' on a null object reference in method 'java.lang.String fr.neamar.kiss.utils.ShortcutUtil.generateShortcutId(fr.neamar.kiss.utils.UserHandle, fr.neamar.kiss.db.ShortcutRecord)'
12-04 08:14:23.887 17946 17946 E AndroidRuntime:        at fr.neamar.kiss.utils.ShortcutUtil.generateShortcutId(r8-map-id-a4b1011dc281e35e290c7c4d75f885a41b6e8d7c7e894904f5c6d961dc63c13d:10)
12-04 08:14:23.887 17946 17946 E AndroidRuntime:        at fr.neamar.kiss.db.DB$ConvertShortcutInfo.<init>(r8-map-id-a4b1011dc281e35e290c7c4d75f885a41b6e8d7c7e894904f5c6d961dc63c13d:5)
12-04 08:14:23.887 17946 17946 E AndroidRuntime:        at fr.neamar.kiss.db.DB$$ExternalSyntheticLambda4.accept(r8-map-id-a4b1011dc281e35e290c7c4d75f885a41b6e8d7c7e894904f5c6d961dc63c13d:32)
12-04 08:14:23.887 17946 17946 E AndroidRuntime:        at fr.neamar.kiss.db.DB$$ExternalSyntheticApiModelOutline0.m(r8-map-id-a4b1011dc281e35e290c7c4d75f885a41b6e8d7c7e894904f5c6d961dc63c13d:1)
12-04 08:14:23.887 17946 17946 E AndroidRuntime:        at fr.neamar.kiss.db.DB.onUpgrade(r8-map-id-a4b1011dc281e35e290c7c4d75f885a41b6e8d7c7e894904f5c6d961dc63c13d:61)
12-04 08:14:23.887 17946 17946 E AndroidRuntime:        at fr.neamar.kiss.db.DBHelper.getDatabase(r8-map-id-a4b1011dc281e35e290c7c4d75f885a41b6e8d7c7e894904f5c6d961dc63c13d:17)
12-04 08:14:23.887 17946 17946 E AndroidRuntime:        at fr.neamar.kiss.TagsHandler.<init>(r8-map-id-a4b1011dc281e35e290c7c4d75f885a41b6e8d7c7e894904f5c6d961dc63c13d:11)
12-04 08:14:23.887 17946 17946 E AndroidRuntime:        at fr.neamar.kiss.DataHandler.getTagsHandler(r8-map-id-a4b1011dc281e35e290c7c4d75f885a41b6e8d7c7e894904f5c6d961dc63c13d:9)
12-04 08:14:23.887 17946 17946 E AndroidRuntime:        at fr.neamar.kiss.loader.LoadAppPojos.<init>(r8-map-id-a4b1011dc281e35e290c7c4d75f885a41b6e8d7c7e894904f5c6d961dc63c13d:18)
12-04 08:14:23.887 17946 17946 E AndroidRuntime:        at fr.neamar.kiss.dataprovider.AppProvider.reload(r8-map-id-a4b1011dc281e35e290c7c4d75f885a41b6e8d7c7e894904f5c6d961dc63c13d:8)
12-04 08:14:23.887 17946 17946 E AndroidRuntime:        at fr.neamar.kiss.dataprovider.Provider.onCreate(r8-map-id-a4b1011dc281e35e290c7c4d75f885a41b6e8d7c7e894904f5c6d961dc63c13d:4)
12-04 08:14:23.887 17946 17946 E AndroidRuntime:        at fr.neamar.kiss.dataprovider.AppProvider.onCreate(r8-map-id-a4b1011dc281e35e290c7c4d75f885a41b6e8d7c7e894904f5c6d961dc63c13d:102)

```

**Additional information**
 - App version: 3.24.0
 - Did the crash start happening with this version? Yes
 - App installed via: F-Droid

I am running LineageOS 23 / Android 16. Please let me know if there is any way I can help test or get more information about what is causing this.

--- a/app/src/main/java/fr/neamar/kiss/db/DB.java
+++ b/app/src/main/java/fr/neamar/kiss/db/DB.java
@@ -10,6 +10,8 @@ import android.preference.PreferenceManager;
 import android.text.TextUtils;
 import android.util.Log;
 
+import androidx.annotation.NonNull;
+
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
@@ -113,10 +115,12 @@ class DB extends SQLiteOpenHelper {
             List<ShortcutInfo> shortcutInfos = ShortcutUtil.getAllShortcuts(mContext);
             List<ConvertShortcutInfo> shortcuts = new ArrayList<>();
             shortcutInfos.forEach(shortcutInfo -> {
-                if (!shortcutInfo.isDynamic() || shortcutInfo.isPinned()) {
-                    ShortcutRecord shortcutRecordWithName = ShortcutUtil.createShortcutRecord(mContext, shortcutInfo, true);
+                ShortcutRecord shortcutRecordWithName = ShortcutUtil.createShortcutRecord(mContext, shortcutInfo, true);
+                if (shortcutRecordWithName != null) {
                     shortcuts.add(new ConvertShortcutInfo(UserHandle.OWNER, shortcutRecordWithName));
-                    ShortcutRecord shortcutRecordWithoutName = ShortcutUtil.createShortcutRecord(mContext, shortcutInfo, false);
+                }
+                ShortcutRecord shortcutRecordWithoutName = ShortcutUtil.createShortcutRecord(mContext, shortcutInfo, false);
+                if (shortcutRecordWithoutName != null) {
                     shortcuts.add(new ConvertShortcutInfo(UserHandle.OWNER, shortcutRecordWithoutName));
                 }
             });
@@ -157,10 +161,10 @@ class DB extends SQLiteOpenHelper {
 
     private static class ConvertShortcutInfo {
 
-        private final String oldId;
-        private final String newId;
+        final String oldId;
+        final String newId;
 
-        private ConvertShortcutInfo(UserHandle userHandle, ShortcutRecord shortcutRecord) {
+        ConvertShortcutInfo(UserHandle userHandle, @NonNull ShortcutRecord shortcutRecord) {
             this.oldId = ShortcutUtil.generateShortcutId(null, shortcutRecord);
             this.newId = ShortcutUtil.generateShortcutId(userHandle, shortcutRecord);
         }

--- a/app/src/main/java/fr/neamar/kiss/utils/ShortcutUtil.java
+++ b/app/src/main/java/fr/neamar/kiss/utils/ShortcutUtil.java
@@ -39,7 +39,7 @@ public class ShortcutUtil {
     /**
      * @return shortcut id generated from shortcut name
      */
-    public static String generateShortcutId(UserHandle userHandle, ShortcutRecord shortcutRecord) {
+    public static String generateShortcutId(UserHandle userHandle, @NonNull ShortcutRecord shortcutRecord) {
         if (userHandle == null) {
             return ShortcutPojo.SCHEME + shortcutRecord.name.toLowerCase(Locale.ROOT);
         } else {
@@ -161,6 +161,7 @@ public class ShortcutUtil {
      * Create ShortcutPojo from ShortcutInfo
      */
     @RequiresApi(Build.VERSION_CODES.O)
+    @Nullable
     public static ShortcutRecord createShortcutRecord(Context context, ShortcutInfo shortcutInfo, boolean includePackageName) {
         if (shortcutInfo.hasKeyFieldsOnly()) {
             // If ShortcutInfo holds only key fields shortcut including data must be fetched

